queries/powershell/highlights.scm

```diff
@@ -0,0 +1,323 @@
+    "System.Management.Automation.ModuleType" "Management.Automation.PSModuleInfo"
+    "System.Management.Automation.PSModuleInfo" "Management.Automation.PSCustomObject"
+    "System.Management.Automation.PSCustomObject" "Management.Automation.PSScriptCmdlet"
+    "System.Management.Automation.PSScriptCmdlet" "Management.Automation.ErrorRecord"
+    "System.Management.Automation.ErrorRecord" "Management.Automation.PSCredential"
+    "System.Management.Automation.PSCredential" "Management.Automation.PSMethod"
+    "System.Management.Automation.PSMethod")
+  )
+
+; Function definitions
+;---------------------
+(function_statement
+  (function_name) @function)
+
+; Classes, fields and methods definitions
+;----------------------------------------
+(class_property_definition
+  (variable) @variable.member)
+
+(class_method_definition
+  (simple_name) @function.method)
+
+(class_statement
+  (simple_name) @_class_name
+  (class_method_definition
+    (simple_name) @constructor)
+  (#eq? @constructor @_class_name))
+
+; Functions, methods calls and properties access
+;--------------------------
+(command_invokation_operator) @operator
+
+(invokation_expression
+  (member_name) @function.call)
+
+(invokation_expression
+  (member_name) @constructor
+  (#eq? @constructor "new"))
+
+(member_access
+  (member_name
+    [
+      (simple_name)
+      (variable)
+    ] @variable.member))
+
+; Commands execution
+;-------------------
+(command
+  command_name: (command_name) @function.call)
+
+(command
+  command_name: (command_name) @function.builtin
+  (#any-of? @function.builtin
+    ; Locations & Paths
+    "cd" "cd.." "cd\\" "cd~" "help" "Get-Location" "Pop-Location" "Push-Location" "Set-Location"
+    "Convert-Path" "Join-Path" "Resolve-Path" "Split-Path" "Test-Path"
+    ; Files & Directories
+    "Clear-Item" "Clear-ItemProperty" "Copy-Item" "Copy-ItemProperty" "Get-Item" "Get-ItemProperty"
+    "Get-ItemPropertyValue" "Invoke-Item" "Move-Item" "Move-ItemProperty" "New-Item"
+    "New-ItemProperty" "Remove-Item" "Remove-ItemProperty" "Rename-Item" "Rename-ItemProperty"
+    "Set-Item" "Set-ItemProperty"
+    ; Aliases
+    "New-Alias" "Get-Alias" "Set-Alias" "Remove-Alias" "Import-Alias" "Export-Alias"
+    ; Host (prompt)
+    "prompt" "cls" "echo" "Clear-Host" "Get-Host" "Out-Host" "Read-Host" "Write-Host" "Out-String"
+    ; Expressions
+    "Invoke-Expression"
+    ; Objects
+    "Compare-Object" "ForEach-Object" "Group-Object" "Measure-Object" "New-Object" "Select-Object"
+    "Sort-Object" "Tee-Object" "Where-Object"
+    ; Modules
+    "Export-ModuleMember" "Find-Module" "Get-Module" "Import-Module" "Install-Module" "New-Module"
+    "New-ModuleManifest" "Publish-Module" "Remove-Module" "Save-Module" "Test-ModuleManifest"
+    "Uninstall-Module" "Update-Module" "Update-ModuleManifest"))
+
+; Parameters and call arguments
+;------------------------------
+(command_elements) @variable.parameter
```

Comment:
I feel like this is unneeded since they are technically arguments and (unlike bash) most arguments already have good highlights (maybe just specify this to `(command_elements (generic_token) @variable.parameter)`?

Reply:
Definitely my mistake, `command_elements` too wide capture, your suggestion is much better, thanks!

Reply:
Thank you! Also the capture has to be within the first parentheses, otherwise it applies to the entire parent capture still. So it would be:
```
(command_elements
  (generic_token) @variable.parameter)
```

Reply:
Whoopsie! Now fixed